recipes/mongo-c-driver/all/conanfile.py

```diff
@@ -0,0 +1,155 @@
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://mongoc.org/"
+    description = "A Cross Platform MongoDB Client Library for C"
+    topics = ("conan", "libbson", "libmongoc", "mongo", "mongodb", "database", "db")
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = ["patches/**"]
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_ssl": [False, 'DARWIN', 'WINDOWS', 'OPENSSL', 'LIBRESSL'],
+        "with_zlib": [False, 'SYSTEM', 'BUNDLED']
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_ssl": 'OPENSSL',
+        "with_zlib": 'BUNDLED'
+    }
+    generators = "cmake"
+    no_copy_source = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "{name}-{version}".format(
+            name=self.name,
+            version=self.version
+        )
+        os.rename(extracted_dir, self._source_subfolder)
+        self._patch_sources()
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_ssl == 'OPENSSL':
+            self.requires("openssl/1.1.1g")
+        if self.options.with_ssl == 'LIBRESSL':
+            self.output.warn("Can be broken. Prefer OpenSSL instead of LIBRESSL")
+            self.requires("libressl/3.2.0")
+        if self.options.with_zlib == 'SYSTEM':
+            self.requires("zlib/1.2.11")
+            if tools.os_info.is_windows:
+                self.output.warn("Usage zlib provided by conan on Windows can be broken")
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="THIRD_PARTY_NOTICES", dst="licenses", src=self._source_subfolder)
+
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["bson-1.0", "mongoc-1.0"] if self.options.shared else \
+            ["bson-static-1.0", "mongoc-static-1.0"]
+        self.cpp_info.includedirs.append(os.path.join("include", "libmongoc-1.0"))
+        self.cpp_info.includedirs.append(os.path.join("include", "libbson-1.0"))
```

Comment:
bson depends on several system libs:
- m, rt, pthread on Linux
- ws2_32 on Windows

https://github.com/mongodb/mongo-c-driver/blob/master/src/libbson/CMakeLists.txt

Reply:
addressed in n-bes#1